odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,318 @@
+        psf : `array` of small size, optional
+            Point Spread Functions to be passed to NiftyRec
+            for the correction of the collimator-detector response
+
+        References
+        ----------
+        .._NiftyRec: http://niftyrec.scienceontheweb.net/wordpress/
+
+        .._NiftyPy: https://github.com/spedemon/niftypy
+
+        """
+        if not isinstance(dom, DiscreteLp):
+            raise TypeError('domain {!r} is not a `DiscreteLp`'
+                            ' instance.'.format(dom))
+
+        if not isinstance(geometry, ParallelHoleCollimatorGeometry):
+            raise TypeError('geometry {!r} is not a'
+                            '`ParallelHoleCollimatorGeometry` instance.'
+                            ''.format(geometry))
+
+        ran = uniform_discr_frompartition(geometry.partition)
+        super().__init__(dom, ran, linear=True)
+
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        if impl.startswith('niftyrec'):
+            if not NIFTYREC_AVAILABLE:
+                raise ValueError('NiftyRec back-end not available.')
+            if impl == 'niftyrec_gpu' and dom.shape != (128, 128, 128):
+                raise NotImplementedError('`nifryrec_gpu` back-end only '
+                                          'supports volume size '
+                                          '(128, 128, 128), got {}'
+                                          ''.format(dom.shape))
+            self._use_gpu = (impl == 'niftyrec_gpu')
+
+        self._geometry = geometry
+
+        psf = kwargs.pop('psf', None)
+        if psf is not None:
+            self._psf = np.asarray(psf)
+        else:
+            self._psf = psf
+        if attenuation is not None:
+            self._attenuation = self.domain.element(attenuation)
+        else:
+            self._attenuation = attenuation
+        self._impl = impl
+
+    def _call(self, x):
+        """Compute the discrete SPECT back-projection.
+
+        Parameters
+        ----------
+        x : `DiscreteLpVector`
+           The volume that is forward projected. The volume is an element
+           in the domain of the operator.
+
+        Returns
+        -------
+        out : ``range`` element
+        """
+        if self._impl.startswith('niftyrec'):
+            return self._call_niftyrec(x.asarray())
+        else:
+            raise NotImplementedError('Only NiftyRec back-end supported '
+                                      'at present')
+
+    def _call_niftyrec(self, x):
+        """Return ``self(x)`` for niftyrec back-end.
+
+        Parameters
+        ----------
+        x :  `numpy.ndarray`
+           The volume that is forward projected. The volume is an element
+           in the domain of the operator.
+
+
+        Returns
+        -------
+        out : ``range`` element
```

Comment:
Doesn't [this](https://github.com/odlgroup/odl/blob/issue-387__spect/odl/tomo/operators/spect_trafo.py#L172) automatically return `out` as an element of `range`...